### PR TITLE
feat(bedrock): add checks for guardrails configuration and log encryption

### PR DESCRIPTION
**Pull request by MrCloudSec**
### Context

This PR addresses security concerns for Amazon Bedrock by introducing three new checks to strengthen protection against prompt attacks, sensitive data exposure, and log security. These checks aim to mitigate the risks associated with generative AI infrastructures being targeted by adversaries.

### Description
This PR adds the following three checks:

- `bedrock_guardrail_prompt_attack_filter_enabled`
Ensures the prompt attack filter is set to the highest strength to protect models from malicious prompts.

- `bedrock_guardrail_sensitive_information_filter_enabled`
Verifies sensitive information filters are enabled to prevent data leakage in model outputs.

- `bedrock_model_invocation_logs_encryption_enabled`
Ensures invocation logs are encrypted with AWS KMS to protect sensitive log data from unauthorized access.

These checks enhance Bedrock's security and data protection.

### Checklist

- Are there new checks included in this PR? Yes / No
    - If so, do we need to update permissions for the provider? Please review this carefully.
- [ ] Review if the code is being covered by tests.
- [ ] Review if code is being documented following this specification https://github.com/google/styleguide/blob/gh-pages/pyguide.md#38-comments-and-docstrings
- [ ] Review if backport is needed.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/providers/aws/services/bedrock/bedrock_service.py [hunk 1]
```diff
@@ -40,8 +44,65 @@ def _get_model_invocation_logging_configuration(self, regional_client):
...
+            guardrail_info = self.regional_clients[guardrail.region].get_guardrail(
+                guardrailIdentifier=guardrail.id
+            )
+            guardrail.sensitive_information_filter = (
+                "sensitiveInformationPolicy" in guardrail_info
+            )
+            for filter in guardrail_info.get("contentPolicy", {}).get("filters", []):
+                if filter.get("type") == "PROMPT_ATTACK":
+                    guardrail.prompt_attack_filter_strength = filter.get(
+                        "inputStrength", None
```
Comment (puchy22) [inline prowler/providers/aws/services/bedrock/bedrock_service.py]:
```suggestion
                        "inputStrength", "NONE"
```

File: tests/providers/aws/services/bedrock/bedrock_guardrail_prompt_attack_filter_enabled/bedrock_guardrail_prompt_attack_filter_enabled_test.py [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+)
+
+make_api_call = botocore.client.BaseClient._make_api_call
+
+GUARDRAIL_ARN = (
+    f"arn:aws:bedrock:{AWS_REGION_US_EAST_1}:{AWS_ACCOUNT_NUMBER}:guardrail/test-id"
+)
+
+
+def mock_make_api_call(self, operation_name, kwarg):
```
Comment (puchy22) [inline tests/providers/aws/services/bedrock/bedrock_guardrail_prompt_attack_filter_enabled/bedrock_guardrail_prompt_attack_filter_enabled_test.py]: I believe that functions should use a name more characteristic of their behaviour than v2 and v3.

File: tests/providers/aws/services/bedrock/bedrock_guardrail_prompt_attack_filter_enabled/bedrock_guardrail_prompt_attack_filter_enabled_test.py [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+                    },
+                ]
+            },
+            "blockedInputMessaging": "Sorry, the model cannot answer this question.",
+            "blockedOutputsMessaging": "Sorry, the model cannot answer this question.",
+        }
+    return make_api_call(self, operation_name, kwarg)
+
+
+def mock_make_api_call_v2(self, operation_name, kwarg):
```
Comment (puchy22) [inline tests/providers/aws/services/bedrock/bedrock_guardrail_prompt_attack_filter_enabled/bedrock_guardrail_prompt_attack_filter_enabled_test.py]:
```suggestion
def mock_make_api_call_no_filters(self, operation_name, kwarg):
```

File: tests/providers/aws/services/bedrock/bedrock_guardrail_prompt_attack_filter_enabled/bedrock_guardrail_prompt_attack_filter_enabled_test.py [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+            "guardrailArn": GUARDRAIL_ARN,
+            "status": "READY",
+            "contentPolicy": {"filters": []},
+            "blockedInputMessaging": "Sorry, the model cannot answer this question.",
+            "blockedOutputsMessaging": "Sorry, the model cannot answer this question.",
+        }
+    return make_api_call(self, operation_name, kwarg)
+
+
+def mock_make_api_call_v3(self, operation_name, kwarg):
```
Comment (puchy22) [inline tests/providers/aws/services/bedrock/bedrock_guardrail_prompt_attack_filter_enabled/bedrock_guardrail_prompt_attack_filter_enabled_test.py]:
```suggestion
def mock_make_api_call_filter_strength_low(self, operation_name, kwarg):
```

File: tests/providers/aws/services/bedrock/bedrock_guardrail_sensitive_information_filter_enabled/bedrock_guardrail_sensitive_information_filter_enabled_test.py [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+)
+
+make_api_call = botocore.client.BaseClient._make_api_call
+
+GUARDRAIL_ARN = (
+    f"arn:aws:bedrock:{AWS_REGION_US_EAST_1}:{AWS_ACCOUNT_NUMBER}:guardrail/test-id"
+)
+
+
+def mock_make_api_call(self, operation_name, kwarg):
```
Comment (puchy22) [inline tests/providers/aws/services/bedrock/bedrock_guardrail_sensitive_information_filter_enabled/bedrock_guardrail_sensitive_information_filter_enabled_test.py]: Please could change all the `mock_make_api_call_vX` in all tests for more descriptive names.

**Review by puchy22 — CHANGES_REQUESTED** [review]

**Comment by MrCloudSec** [comment]
@puchy22, changes done in https://github.com/prowler-cloud/prowler/pull/5385/commits/a50114f41424d69ff72472849b896ccd8ba27d89

**Review by puchy22 — APPROVED** [review]
Awesome!! 💯

**Pull request merged**